lib/api.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const setupContext = require('./context');
+const Widget = require('./widget');
+const _ = require('lodash');
+
+module.exports = {createClient};
+
+function createClient (options) {
+  options = options || {};
+  const context = setupContext({
+    token: options.accessToken,
+    host: options.host
+  });
+
+  return {get: get, getAll, save, delete: del};
+
+  function get (id) {
```

Comment:
Why this pattern in having functions within the outer function and the `return` statement before them?